### ISSUE TITLE
Improve on a symbolic function

### ISSUE THREAD

**Issue by AmitMY** [KindXiaoming/pykan#79]
I'm trying to optimize a heuristic someone wrote manually, with gold data I have.
Right now, I'm training an MLP, but it would be nice to create a KAN if I could start it off from the heuristic -

The heuristic is, given (x1, y1, z1), (x2, y2, z2), (x3, y3, z3), calculate:
`5.4*\sqrt{(x_3 - \frac{2*x_1 + x_2}{3})^2 + (y_3 - \frac{2*y_1 + y_2}{3})^2}`
![image](https://github.com/KindXiaoming/pykan/assets/5757359/3e738939-e6fe-4989-b388-4988f5afdb41)

**Comment by KindXiaoming** [comment]
Wow thanks for sharing the use case! If I understand correctly, the problem is essentially compiling a symbolic formula into a KAN?

**Comment by AmitMY** [comment]
yes! compiling the formula into a (pruned) KAN so that I can then optimize the KAN to find use for my extra inputs (for example, here I have a `z` axis that is unused by the formula)

**Comment by AmitMY** [comment]
Any update on this @KindXiaoming ? Would love to be able to init like this

**Comment by KindXiaoming** [comment]
@AmitMY  Introducing KAN compiler :)

Tutorial: https://github.com/KindXiaoming/pykan/blob/master/tutorials/Interp_3_KAN_Compiler.ipynb
